pycloudlib/azure/instance.py

```diff
@@ -241,8 +244,14 @@ def add_network_interface(self) -> str:
         """Add network interface to instance.
 
         Creates NIC and adds to the VM instance.
+
+        NOTE: It will deallocate the virtual machine, add the NIC,
+        then start the virtual machine.
+
         Returns the ip address of the new NIC.
         """
+        # NOTE: Azure needs the VM to be deallocated to add NICs
+        self.deallocate()
```

Comment:
I think deallocate / start is not needed in this function, as it is already performed in _attach_nic_to_vm. Or am I missing something?

Reply:
Oh yeah, I had not picked up on it.